### ISSUE TITLE
Hide underground parkings

### ISSUE THREAD

**Issue by Tomasz-W** [gravitystorm/openstreetmap-carto#3506]
Continuation of https://github.com/gravitystorm/openstreetmap-carto/issues/2904 discussion since https://github.com/gravitystorm/openstreetmap-carto/issues/2904#issuecomment-346001797.

Proposition is to render underground parkings (`amenity=parking` + `parking=underground`) only as icon of "P" with down arrow.

Test renderings provided by @kocio-pl : 

Before:
![](https://user-images.githubusercontent.com/5439713/33073892-9054e888-cec4-11e7-887f-d2dbb7971974.png)

After:
![](https://user-images.githubusercontent.com/5439713/45931889-cb234d00-bf74-11e8-8298-ac77e1eabeab.png)

https://www.openstreetmap.org/way/235550028

Gist link: https://gist.github.com/Tomasz-W/b9655543be00a01d51c975b03009c654 -> v5

**Comment by polarbearing** [comment]
Yes it makes sense to have a separate icon. 
I am trying to understand how rendering the area was done in the 'before' situation. 
I see some white area changing to green between before/after, but that does not match the outline I see from the way link?

**Comment by kocio-pl** [comment]
Before parking color change this was a yellow all around - see https://github.com/gravitystorm/openstreetmap-carto/issues/2904#issuecomment-346001797 - so now this is parking grey visible:

http://www.openstreetmap.org/way/230629285#map=18/41.88262/-87.62070
![screenshot-2017-11-21 openstreetmap linia millennium lakeside garage 230629285](https://user-images.githubusercontent.com/5439713/33073882-895d1d5c-cec4-11e7-94a9-916271ab9914.png)

**Comment by jragusa** [comment]
It would be better to even hide this icon for 2 reasons:
1/ it's more important for an underground parking to know where is the entrance (for both pedestrian and cars) than the "centered" location because access are more restricted than for the surface parking.
2/ we will have 2 different icons related to underground parking: this one and those for the entrance (#3505)

**Comment by kocio-pl** [comment]
I am also still not sure if we need underground parking icon at all.

By the way - there are more things that we probably should hide for underground parkings:

https://www.openstreetmap.org/relation/8452071#map=18/55.75128/37.62905

![screenshot_2018-11-10 openstreetmap](https://user-images.githubusercontent.com/5439713/48304352-c1e84280-e518-11e8-9a50-503bbca6f9e8.png)

**Comment by kocio-pl** [comment]
Also related to #552 - and I think we should hide underground buildings.

**Comment by kocio-pl** [comment]
> I hope you remember that I was against rendering parking_space in general ;-)

Still guilty - you were not against rendering "P" letter and parking aisles... :smile: 

> OK but I still don't understand why the area south of the word "Jay Pritzker Pavillion" changes from grey to green in the before/after. Why wasn't it green before?

That's probably because somebody made a small "park" area (it might be a tagging mistake, since the park is probably much larger), which covers all bigger areas:

https://www.openstreetmap.org/way/470069404#map=18/41.88350/-87.62233

**Comment by jeisenbe** [comment]
I agree that underground parking should just have the entrance rendered.

**Comment by Tomasz-W** [comment]
I propose to wait for https://github.com/gravitystorm/openstreetmap-carto/pull/3505 merge and then check if rendering just parking entrances would be enough.

**Comment by kocio-pl** [comment]
I have just merged #3505. I'm inclined toward removing underground parking rendering, including icon, area, roads and individual parking places.

**Comment by Adamant36** [comment]
Underground roads might be important for places where its going through a tunnel to and from the parking. Outside of that though, I agree the other things could probably be removed.

**Comment by kocio-pl** [comment]
Tagging can be different, for example here is just `highway=service`, but probably it should be tagged as a parking aisle, and we could hide all the parking aisles with `location=underground` and `tunnel=yes`:

https://www.openstreetmap.org/way/340493731#map=19/50.05602/19.93262

![screenshot_2018-11-13 openstreetmap](https://user-images.githubusercontent.com/5439713/48407639-9c854f80-e737-11e8-91ef-7e18712f9d42.png)

**Comment by Adamant36** [comment]
That would work. People don't seem to like using the service road sub tags much because of how thin they render. So whatever we can do to encourage more correct tagging of them.

**Comment by polarbearing** [comment]
> should be tagged as a parking aisle

No, the access to a parking area is _not_ supposed to be tagged parking_aisle, though this is a common mistake. 

A tunnel is by definition underground, and we cannot drop tunnel rendering.

**Comment by polarbearing** [comment]
With the tendency to indoor mapping, this is probably on the increase.

**Comment by Adamant36** [comment]
102,219 service roads are tagged with the tunnel tag. Of those, 52,833 are tagged as tunnel=building_passage. So, although the tunnel tag on service roads isn't that prevalent compared to how many there are overall, currently 22,726, 431, a good number number of them are tagged as going under a building. I guess to continue rendering them or not depends on if your going with how many service are mapped underground or how many are mapped mapped as a percentage of the total overall. Personally, I prefer @Tomasz-W's way of mapping them where they end at the building, but that doesn't seem to be how they are being mapped.

**Comment by kocio-pl** [comment]
I'm not sure if I was clear - I meant dropping rendering only "parking aisles with location=underground and tunnel=yes" ( https://wiki.openstreetmap.org/wiki/Tag:service%3Dparking_aisle ), and not _any_ road with location=underground and tunnel=yes.

**Comment by polarbearing** [comment]
Then the main service way, which should be without 'aisle', would remain visible in those underground parkings, which might create confusion. I also doubt that tunnel=yes is correct tagging for underground parking; shouldn't there be a negative level number?

**Comment by jragusa** [comment]
@polarbearing yes, only `location=underground` and `level=*` should be tagged. `tunnel=yes` is useless here excepted for access way.

**Comment by jragusa** [comment]
![parking_underground_after](https://user-images.githubusercontent.com/9897203/50363526-560fe580-056c-11e9-8fe5-3f4222d69b40.png)

Once merged, this modification will lead to salutary corrections of nodes erroneously tagged with `amenity=parking` + `parking=underground` to `amenity=parking_entrance`.

**Comment by kocio-pl** [comment]
Reopening, since there are still more underground parking objects to be hidden - see https://github.com/gravitystorm/openstreetmap-carto/issues/3506#issuecomment-437602119 and https://github.com/gravitystorm/openstreetmap-carto/issues/3506#issuecomment-438216801 for example.

**Comment by jragusa** [comment]
For the first location, I don't think we can do anything except removing the underground service highway in this case. Removing `highway=service` with `layer=-1` would be cause some side effects elsewhere.

For the second location, we can restrict to amenity=parking_space AND (parking=surface OR parking=null). About `highway=service`, `indoor=yes` tag could help.

**Comment by BertMule** [comment]
Please hide the damn thing.
Only the entrance is relevant, with a clear marking (and not hidden when private or anything #https://github.com/gravitystorm/openstreetmap-carto/issues/4890).

Currently this **underground** garage hides an entire park on top of it.
Boundaries are questionable, but probably imported automatically.
[Sacharovplaats](https://www.openstreetmap.org/#map=19/51.90755/4.33983)

It is also repeatedly disappointing that this issue from **2018** is still left unfinished.

**Comment by pnorman** [comment]
> It is also repeatedly disappointing that this issue from **2018** is still left unfinished.

No one has proposed a pull request for it, so someone would need to decide that it is a priority for them and write a PR. I expect all maintainers view other issues as more important so it's unlikely to be one of us, but it doesn't need to be.

As a reminder, among the [values of the Code of Conduct](https://github.com/gravitystorm/openstreetmap-carto/blob/master/CODE_OF_CONDUCT.md#values) is friendly and welcoming.

**Comment by imagico** [comment]
Expressing a demanding attitude towards volunteer developers here is not appropriate.

Apart from that the comment is off-topic and factually incorrect.  This issue is now (after #3600) about the idea to hide elements of underground parkings (like individually mapped parking spaces).  I am editing the title to reflect that.
